### PR TITLE
OOP school library: set up associations

### PR DESCRIPTION
**Pull request by RashidMenhas**
# Project requirements

## Create a class Classroom with the following:
- @label instance variable, should be initialized in the constructor.
- Setter and getter for @label (remember about attr_accessor).

#### Create the has-many/belongs-to relationship between Classroom and Student. The following should be implemented:
- Create the has-many side (a classroom has many students).
- Create the belongs-to side (a student belongs to a classroom).

## Create a class Book with the following:
- @title and @author instance variables, should be initialized in the constructor.
- Setters and getters for instance variables (remember about attr_accessor).

## Create a class Rental with the following:
- @date instance variable, should be initialized in the constructor.
- Setter and getter for @date (remember about attr_accessor).

#### Create the many-to-many (also has-many-through) relationship between Person and Book using the intermediate class Rental. The following should be implemented:
- Create the has-many side of Book and Rental (a book has many rentals).
- Create the belongs-to side of Rental and Book (a rental belongs to a book).
- Create the has-many side of Person and Rental (a person has many rentals).
- Create the belongs-to side of Rental and Person (a rental belongs to a person).
- Modify the constructor of Rental so Book and Person are set in it.

### CODE REVIEW

File: book.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
+class Book
+  attr_accessor :title, :rentals, :author
+
+  def initialize(title, author)
+    @title = title
+    @author = author
+    @rentals = []
+  end
+
+  def add_rental(person, date)
```
Comment (GzimAsani) [inline book.rb:10]: - [ ] Add self as a parameter on `add_rental(person, self, date)`

File: person.rb [hunk 2]
```diff
@@ -0,0 +1,35 @@
+require './nameable'
+
+class Person < Nameable
+  attr_reader :id, :rentals
+  attr_accessor :name, :age
```
Comment (GzimAsani) [inline person.rb]: - [ ] :rentals needs to be getter and setter in this case therefore please add it under attr_accessor

File: README.md [hunk 3]
```diff
@@ -1,2 +1,206 @@
...
+  - [⭐️ Show your support ](#️-show-your-support-)
+  - [🙏 Acknowledgments ](#-acknowledgments-)
+  - [📝 License ](#-license-)
+
+<!-- PROJECT DESCRIPTION -->
+
+
+# 📖 oop-school-library <a name="about-project"></a>
+
+> oop-school-library is the implementation of oop in ruby language with person super-class and student, teacher as sub-class.
```
Comment (GzimAsani) [inline README.md]: - [ ] Remove descriptions from the readme file, they're there for you to understand what to write 
image

**Review by GzimAsani — CHANGES_REQUESTED** [review]
### Hello @RashidMenhas                     :wave:   
 Good job so far! There are some issues that you still need to work on to go to the next project but you are almost there!
 
 ## Highlights
 * Good job with associations, truly a great job! You only have few changes to make  :+1: 

 ## Required Changes 
- [ ] Check the comments below :arrow_double_down: 




 ## Optional suggestions
 _Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._



Cheers and Happy coding! clapclapclap
 
 Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.
 
 **_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**
 
 _As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Nemwel-Boniface — APPROVED** [review]
## Hello @RashidMenhas          :wave: ,
I am Nemwel your reviewer for this PR :man_beard:  

Thank you for meeting all the project requirements good job! :partying_face: 
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! :partying_face: 🎉 



## Highlights

Great job setting up the correct linters ✔️ 
Professional documentation ✔️ 
Correct workflow ✔️ 
Meaningful commit messages ✔️ 
Met all the milestone requirements ✔️ 

## [Optional] suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me **@Nemwel-Boniface** in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by RashidMenhas** [comment]
brother please help me i face a problem on day 4 and i spent 2 days on it but didn't fix it please help me I will be very grateful to you it's about  (execution permission).Sent from my Verizon, Samsung Galaxy smartphone
-------- Original message --------From: Gzim Asani ***@***.***> Date: 1/20/23  2:05 AM  (GMT+05:00) To: RashidMenhas/oop-school-library ***@***.***> Cc: Rashid Menhas ***@***.***>, Mention ***@***.***> Subject: Re: [RashidMenhas/oop-school-library] OOP school library: set up associations (PR #3) 

@GzimAsani requested changes on this pull request.



Hello @RashidMenhas                     👋

Good job so far! There are some issues that you still need to work on to go to the next project but you are almost there!

Highlights



Good job with associations, truly a great job! You only have few changes to make  👍



Required Changes



 Check the comments below ⏬



Optional suggestions

Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.

Cheers and Happy coding! clapclapclap

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.

Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.

As described in the Code reviews limits policy you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using this form.



In book.rb:

> @@ -0,0 +1,13 @@

+class Book

+  attr_accessor :title, :rentals, :author

+

+  def initialize(title, author)

+    @title = title

+    @author = author

+    @rentals = []

+  end

+

+  def add_rental(person, date)





 Add self as a parameter on add_rental(person, self, date)









In person.rb:

> @@ -0,0 +1,35 @@

+require './nameable'

+

+class Person < Nameable

+  attr_reader :id, :rentals

+  attr_accessor :name, :age





 :rentals needs to be getter and setter in this case therefore please add it under attr_accessor









In README.md:

> +    - [Usage](#usage)

+    - [Run tests](#run-tests)

+    - [Deployment](#deployment)

+  - [👥 Authors ](#-authors-)

+  - [🔭 Future Features ](#-future-features-)

+  - [🤝 Contributing ](#-contributing-)

+  - [⭐️ Show your support ](#️-show-your-support-)

+  - [🙏 Acknowledgments ](#-acknowledgments-)

+  - [📝 License ](#-license-)

+

+

+

+

+# 📖 oop-school-library <a name="about-project"></a>

+

+> oop-school-library is the implementation of oop in ruby language with person super-class and student, teacher as sub-class.





 Remove descriptions from the readme file, they're there for you to understand what to write







—Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you were mentioned.Message ID: ***@***.***>

[

{

***@***.***": "http://schema.org",

***@***.***": "EmailMessage",

"potentialAction": {

***@***.***": "ViewAction",

"target": "https://github.com/RashidMenhas/oop-school-library/pull/3#pullrequestreview-1262594374",

"url": "https://github.com/RashidMenhas/oop-school-library/pull/3#pullrequestreview-1262594374",

"name": "View Pull Request"

},

"description": "View this Pull Request on GitHub",

"publisher": {

***@***.***": "Organization",

"name": "GitHub",

"url": "https://github.com"

}

}

]

**Pull request merged**